code/modules/assembly/doorcontrol.dm

```diff
@@ -1,17 +1,29 @@
+var/global/list/restricted_ids = list() //List of IDs that cannot be custom-set to - the list is added to in buttons.dm
+
 /obj/item/device/assembly/control
 	name = "blast door controller"
 	desc = "A small electronic device able to control a blast door remotely."
 	icon_state = "control"
 	origin_tech = "magnets=1;programming=2"
 	attachable = 1
-	var/id = null
+	var/id = "generic"
 	var/can_change_id = 0
 
 /obj/item/device/assembly/control/examine(mob/user)
 	..()
 	if(id)
 		user << "It's channel ID is '[id]'."
 
+/obj/item/device/assembly/control/interact(mob/user)
+	if(is_secured(user) && can_change_id)
+		var/new_id = stripped_input(user,"Please input a new ID for this assembly.","Set ID", max_length = MAX_NAME_LEN) as null|text
```

Comment:
You forgot to verify if the `mob/user` is still near the `src` after `input()` completes